pkg/maps/lbmap/lbmap.go

```diff
@@ -531,6 +539,13 @@ func (*LBBPFMap) DumpServiceMaps() ([]*loadbalancer.SVC, []error) {
 		}
 	}
 
+	for _, svcKey := range wildcardServiceKeys {
+		if err := deleteServiceLocked(svcKey); err != nil {
+			log.WithField(logfields.ServiceKey, svcKey).
+				WithError(err).Warn("Unable to delete wildcard service entry from BPF map")
+		}
+	}
```

Comment:
Do we need this explicit delete here? My understanding is that when the agent comes up, it restores the agent view based on the BPF map in 1.17 and then it syncs with kube-apiserver.. whatever is left-over gets auto-cleaned from the BPF map then. So my understanding is we don't need this logic here, and above we can just probe for these keys and skip adding them to inconsistentServiceKeys. Did you try this?